app/controllers/ChoiceController.scala

```diff
@@ -19,21 +19,66 @@ package controllers
 import forms.Choice
 import javax.inject.{Inject, Singleton}
 import play.api.i18n.I18nSupport
-import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
-import controllers.actions.AuthenticatedAction
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents, Result, Results}
+import controllers.actions.{AuthenticatedAction, JourneyRefiner}
+import controllers.exchanges.{AuthenticatedRequest, JourneyRequest}
+import models.cache.{Answers, ArrivalAnswers, AssociateUcrAnswers, Cache, DepartureAnswers, DissociateUcrAnswers, ShutMucrAnswers}
+import repositories.MovementRepository
 import views.html.choice_page
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Failure
 
 @Singleton
-class ChoiceController @Inject()(authenticate: AuthenticatedAction, mcc: MessagesControllerComponents, choicePage: choice_page)(
-  implicit ec: ExecutionContext
-) extends FrontendController(mcc) with I18nSupport {
+class ChoiceController @Inject()(
+  authenticate: AuthenticatedAction,
+  getJourney: JourneyRefiner,
+  mcc: MessagesControllerComponents,
+  movementRepository: MovementRepository,
+  choicePage: choice_page
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
 
   def displayChoiceForm(): Action[AnyContent] = authenticate.async { implicit request =>
-    Future.successful(Ok(choicePage(Choice.form)))
+  println("#### " + request.operator.pid)
+    movementRepository.findByPid(request.operator.pid).map {
+      case Some(cache) =>
+        Ok(choicePage(Choice.form().fill(Choice(cache.answers.`type`))))
+      case None        =>
+        Ok(choicePage(Choice.form()))
+    }.recover {
```

Comment:
I'd let this bubble up to the error handler

Reply:
It was just for testing